synapse/rest/media/v1/media_repository.py

```diff
@@ -348,15 +350,30 @@ async def _get_remote_media_impl(
             media_info = await self._download_remote_file(
                 server_name,
                 media_id,
+                file_info,
             )
         except SynapseError:
             raise
         except Exception as e:
             # An exception may be because we downloaded media in another
             # process, so let's check if we magically have the media.
             media_info = await self.store.get_cached_remote_media(server_name, media_id)
-            if not media_info:
+
+            is_cached = False
+
+            if media_info is not None:
+                try:
+                    await self.media_storage.ensure_media_is_in_local_cache(
+                        FileInfo(server_name, media_info["filesystem_id"])
+                    )
+                    is_cached = True
+                except NotFoundError:
+                    pass
+
+            if not media_info or (media_info and not is_cached):
                 raise e
+            else:
+                logger.warning("Ignoring _download_remote_file exception", exc_info=e)
```

Comment:
Looks accurate, but i'll add a bit more verbose commenting;

```suggestion
            if not media_info:
                # We haven't ever cached this, 
                # we just got an error upon downloading, raise.
                raise e
            
            is_cached = False
            try:
                await self.media_storage.ensure_media_is_in_local_cache(
                    FileInfo(server_name, media_info["filesystem_id"])
                )
                is_cached = True
            except NotFoundError:
                pass

            if not is_cached:
                # We cached this previously, but re-download failed, raise.
                raise e
            
            logger.warning("Ignoring _download_remote_file exception because file is present in local filesystem cache", exc_info=e)
```